arm-software/linux/docs/PortingFromACfL.md

```diff
@@ -0,0 +1,101 @@
+## ACfL to ATfL Porting Guide
+
+This document outlines the key differences between using the Arm Toolchain for
+Linux (ATfL) and the Arm Compiler for Linux (ACfL). It also lists command-line
+options and macros that can be used with ATfL as alternatives to those used in
+ACfL.
+
+Both toolchains provide frontends for compiling C, C++, and Fortran code, and
+the names of these frontends are consistent across both: `armclang` for C,
+`armclang++` for C++, and `armflang` for Fortran.
+
+### Reference Version
+
+|Compiler                      |Version|
+|------------------------------|-------|
+|Arm Compiler for Linux (ACfL) |24.10  |
+|Arm Toolchain for Linux (ATfL)|20.0   |
+
+### ArmPL integration
+
+In ACfL, Arm Performance Libraries (ArmPL) are bundled with the compiler. To
+simplify usage, the `-armpl` flag is provided to automatically include the
+necessary headers and libraries for BLAS, LAPACK, FFT, and other numerical
+routines.
+
+In contrast, ATfL does not include ArmPL directly, but it depends on the ArmPL
+package, which is installed alongside the toolchain. To use ArmPL with ATfL, the
+users must manually specify the locations of the headers and libraries using
+`pkg-config`. Detailed instructions are available in the
```

Comment:
I don't think the armpl docs recommend using `pkg-config` but rather provide the flags manually. E.g. see https://developer.arm.com/documentation/101004/2410/General-information/Access-Arm-Performance-Libraries where the flags are just listed by hand.
I think we should probably follow their recommendation rather than making our own?

Reply:
@paulwalker-arm could you comment on that? Using `pkg-config` was something you recommended and everyone seemed to agree, apparently this is the first (and only?) place where it is actually documented. Also, Will has expressed his acceptance of this document while it was on Confluence, so I though the agreement is common.

Reply:
Using `pkg-config` represents best practice but I don't own the toolchain or library so it is not for me to dictate.

Reply:
@DavidTruby is this a satisfying response for you?

Reply:
I don't really have any strong opinions about the actual usage, I just think it odd that we're recommending something than the ArmPL docs here. I'd rather they were aligned because otherwise users might get confused why there's one recommendation here and a different one there. But I'm happy to be overruled on that, it's not a strongly held view 😅